homeassistant/components/goalzero/config_flow.py

```diff
@@ -20,32 +25,60 @@ class GoalZeroFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
+    def __init__(self):
+        """Initialize a Goal Zero Yeti flow."""
+        self.ip_address = None
+
+    async def async_step_dhcp(self, discovery_info):
+        """Handle dhcp discovery."""
+        self.ip_address = discovery_info[IP_ADDRESS]
+
+        await self.async_set_unique_id(discovery_info[MAC_ADDRESS])
+        self._abort_if_unique_id_configured(updates={CONF_HOST: self.ip_address})
```

Comment:
If we can set the same unique id when they manually configure the device this would improve the ux.

Reply:
Is there any way to get the `MAC_ADDRESS` from a device other than pulling it from the `/sysinfo` endpoint?

Reply:
See latest commit.

Reply:
> Is there any way to get the `MAC_ADDRESS` from this device other than pulling it from the `/sysinfo` endpoint?

That's likely the best way.  You should pass it though `format_mac` as well to make sure it matches the same format as dhcp

Reply:
`from homeassistant.helpers.device_registry import format_mac`